econml/dowhy.py

```diff
@@ -0,0 +1,226 @@
+        T: vector of length n
+            Treatments for each sample
+        X: optional (n, d_x) matrix (Default=None)
+            Features for each sample
+        W: optional (n, d_w) matrix (Default=None)
+            Controls for each sample
+        Z: optional (n, d_z) matrix (Default=None)
+            Instruments for each sample
+        outcome_names: optional list (Default=None)
+            Name of the outcome
+        treatment_names: optional list (Default=None)
+            Name of the treatment
+        feature_names: optional list (Default=None)
+            Name of the features
+        confounder_names: optional list (Default=None)
+            Name of the confounders
+        instrument_names: optional list (Default=None)
+            Name of the instruments
+        graph: optional
+            Path to DOT file containing a DAG or a string containing a DAG specification in DOT format
+        estimand_type: optional string
+            Type of estimand requested (currently only "nonparametric-ate" is supported).
+            In the future, may support other specific parametric forms of identification
+        proceed_when_unidentifiable: optional bool (Default=True)
+            Whether the identification should proceed by ignoring potential unobserved confounders
+        missing_nodes_as_confounders: optional bool (Default=False)
+            Whether variables in the dataframe that are not included in the causal graph should be automatically
+            included as confounder nodes
+        control_value: optional scalar (Default=0)
+            Value of the treatment in the control group, for effect estimation
+        treatment_value: optional scalar (Default=1)
+            Value of the treatment in the treated group, for effect estimation
+        target_units: optional (Default="ate")
+            The units for which the treatment effect should be estimated.
+            This can be of three types:
+                (1) a string for common specifications of target units (namely, "ate", "att" and "atc"),
+                (2) a lambda function that can be used as an index for the data (pandas DataFrame),
+                (3) a new DataFrame that contains values of the effect_modifiers and effect will be estimated
+                 only for this new data
+        kwargs: optional
+            Other keyword arguments from fit method for CATE estimator
+
+        Returns
+        -------
+        self
+
+        """
+
+        Y, T, X, W, Z = check_input_arrays(Y, T, X, W, Z)
+
+        # create dataframe
+        n_obs = Y.shape[0]
+        Y, T, X, W, Z = reshape_arrays_2dim(n_obs, Y, T, X, W, Z)
+
+        # currently dowhy only support single outcome and single treatment
+        assert Y.shape[1] == 1, "Can only accept single dimensional outcome."
+        assert T.shape[1] == 1, "Can only accept single dimensional treatment."
+
+        # column names
+        if outcome_names is None:
```

Comment:
WE should be using @moprescu  `cate_outcome_names`

Reply:
Similarly for all the other names below, so that default names are uniformized

Reply:
Yes, this part is pending to change once Miruna's PR is in.

Reply:
I guess I should use the helper function `get_input_columns` directly since here I need the input name of raw data, for example cate_feature_name will give me name after featurizer, and cate_treatment_name will give me the name after OHE if by any chance user calls fit first.

Reply:
done